crates/ruff/src/rules/flake8_return/rules.rs

```diff
@@ -363,29 +357,27 @@ fn has_refs_before_next_assign(id: &str, return_location: Location, stack: &Stac
 fn has_refs_or_assigns_within_try_or_loop(id: &str, stack: &Stack) -> bool {
     if let Some(references) = stack.references.get(&id) {
         for location in references {
-            for (try_location, try_end_location) in &stack.tries {
-                if try_location.row() < location.row() && location.row() <= try_end_location.row() {
+            for try_range in &stack.tries {
```

Comment:
Future if this turns out to be slow: Use a binary search to find the right range.